sdk/java-sdk-protobuf/src/main/scala/kalix/javasdk/impl/Contexts.scala

```diff
@@ -35,6 +36,9 @@ private[impl] trait ActivatableContext extends Context {
  */
 private[kalix] trait InternalContext {
   def getComponentGrpcClient[T](serviceClass: Class[T]): T
+
+  /** Meant to be used by component calls, initially such they have access to info like trace parent */
+  def componentCallMetadata: MetadataImpl = MetadataImpl.Empty
```

Comment:
You mean, following the example, that in the `ScalaActionContextAdapter` would be empty and it shouldn't ? That is, I'm erasing the metadata in case there was some?

Reply:
For the record we had a quick call about this.

Reply:
As a take away: I will initially prove that the current solution is not working for the Scala SDK, although it works for the Java SDK. 
Then I'll leave abstract `def componentCallMetadata: MetadataImpl` in `InternalContext` and then implement it in each of its descendants, accordingly.